src/Sql/Sql/Instance Failover Group/Cmdlet/NewAzureSqlInstanceFailoverGroup.cs

```diff
@@ -0,0 +1,177 @@
+        SupportsShouldProcess = true), OutputType(typeof(AzureSqlInstanceFailoverGroupModel))]
+    public class NewAzureSqlInstanceFailoverGroup : AzureSqlInstanceFailoverGroupCmdletBase
+    {
+        /// <summary>
+        /// Gets or sets the name of the Instance Failover Group to create.
+        /// </summary>
+        [Parameter(Mandatory = true,
+            Position = 2,
+            HelpMessage = "The name of the Azure SQL Database Failover Group to create.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets the partner resource group name for Instance Failover Group
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The name of the secondary resource group of the Instance Failover Group.")]
+        [ValidateNotNullOrEmpty]
+        public string PartnerResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the partner region for Instance Failover Group
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "The name of the partner region of the Instance Failover Group.")]
+        [ValidateNotNullOrEmpty]
+        public string PartnerRegion { get; set; }
+        
+        /// <summary>
+        /// Gets or sets the name of the Managed Instance in the local region
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "The name of the Managed Instance in the local region to be added to the Instance Failover Group.")]
+        [ValidateNotNullOrEmpty]
+        public string PrimaryManagedInstanceName { get; set; }
+        
+        /// <summary>
+        /// Gets or sets the name of the Managed Instance in the partner region
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "The name of the Managed Instance in the partner region to be added to the Instance Failover Group.")]
+        [ValidateNotNullOrEmpty]
+        public string PartnerManagedInstanceName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the partner subscription id for Instance Failover Group
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The subscription id of the secondary managed instance of the Instance Failover Group needed only for cross-subscrption setup")]
+        [ValidateNotNullOrEmpty]
+        public string PartnerSubscriptionId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the failover policy without data loss for the Sql Azure Instance Failover Group.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The failover policy of the Instance Failover Group.")]
+        [ValidateNotNullOrEmpty]
+        [PSDefaultValue(Help = "Automatic")]
+        public FailoverPolicy FailoverPolicy { get; set; }
```

Comment:
@ziwa-msft rather than using an enum to represent a parameter, we advise using our [`PSArgumentCompleter`](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/design-guidelines/parameter-best-practices.md#generic-argument-completer) attribute, which has the same tabbing functionality as enums, but let's users provide values that are outside of the provided range in the case that the API is updated to support new values but the values haven't been added to the range in PowerShell.

This comment applies to all parameters represented as an enum in this PR. We can discuss offline about this if needed.

Reply:
make sense. I have updated the cmdlets to use PSArgumentCompleter instead.